include/mlir/Conversion/GPUToROCM/GPUToROCMPass.h

```diff
@@ -0,0 +1,89 @@
+//===- GPUToROCmPass.h - MLIR ROCm runtime support --------------*- C++ -*-===//
+//
+// Copyright 2019 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+#ifndef MLIR_CONVERSION_GPUTOROCM_GPUTOROCMPASS_H_
+#define MLIR_CONVERSION_GPUTOROCM_GPUTOROCMPASS_H_
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace mlir {
+
+namespace rocm {
+
+/// string constants used by the ROCM backend
+static constexpr const char *kHSACOAnnotation = "amdgpu.hsaco";
+static constexpr const char *kHSACOGetterAnnotation = "amdgpu.hsacogetter";
+static constexpr const char *kHSACOGetterSuffix = "_hsaco";
+static constexpr const char *kHSACOStorageSuffix = "_hsaco_cst";
+
+/// enum to represent the AMD GPU versions supported by the ROCM backend
+enum class AMDGPUVersion { GFX900 };
+
+/// enum to represent the HSA Code Object versions supported by the ROCM backend
+enum class HSACOVersion { V3 };
+
+/// Configurable parameters for generating the HSACO blobs from GPU Kernels
+struct HSACOGeneratorConfig {
+
+  /// Constructor - sets the default values for the configurable parameters
+  HSACOGeneratorConfig(bool isTestMode)
+      : testMode(isTestMode), amdgpuVersion(AMDGPUVersion::GFX900),
+        hsacoVersion(HSACOVersion::V3), rocdlDir("/opt/rocm/lib/"),
+        linkerPath("/opt/rocm/hcc/bin/ld.lld") {}
```

Comment:
I suspect default values like this one should be generated by the cmake build script